metrics/mobile-app-metrics.md

```diff
@@ -0,0 +1,131 @@
+
+![service diagram](./app-service-diagram.jpg)
+
+### Mobile App Client
+
+A native, platform-specific SDK for targetting the Metrics Service will be available to developers.
+
+Upon installation into a project and providing configuration for targetting the Metrics Service the SDK will automatically send general information about the general runtime for this application with a unique device ID, namely the `[SDK Version, clientID, timestamp]` tuple upon application initialization.
+
+The SDK will also instrument calls at the network level in order to gather metrics around communication and bandwith consumption.
+
+Other system resources utilization data can also be sent periodically.
+
+### Mobile Metrics Service
+
+A thin HTTP-based service will be present to receive metrics data from the mobile applications in an storage-agnostic way, then forwarding to the Storage service.
+
+Compared to mobile applications interfacing directly with the Storage service, this presents the following tradeoff:
+
+#### Pros
+
+- Possibility to upgrade or change the Storage service independently of mobile client SDK version and implementation;
+
+- Possible support for different Storage engines which can be selected based on application needs;
+
+#### Cons
+
+- Extra service deployment and scaling needs;
+
+- Development and maintenance of the new service;
+
+- Types of gathered metrics can become a subset of the underlying Storage's full range of types.
+
+<!-- TODO: mention logstash, fluentd -->
+
+### Storage
+
+The Storage service will store metrics data forwarded from the Mobile Metrics Service. Its implementation can be supplied by the following projects:
+
+#### Prometheus
+
+##### Pros
+
+- Shared service with backend metrics
+- Requires an additional service for long-term metrics storage
+
+##### Cons
+
+- Data model is time-series exclusive
+- Pull-based data gathering
+
+#### ElasticSearch
+
+##### Pros
+
+- High advertised performance
+- Highest variety of data types and modelling
+- Rich querying possibilities
+
+##### Cons
+
+- High minimum resource requirements (over evaluation-sized clusters)
+
+#### MongoDB
+
+<!-- TODO -->
+
+##### Pros
+
+##### Cons
+
+#### InfluxDB
+
+##### Pros
+
+- Long-term storage adapter for Prometheus
+- HTTP-based API for push-based metrics gathering
+- Flexible querying with direct comparison to SQL
+
+##### Cons
+
+- Focused on time-series data
+- No clustering support in OSS offering
+- Medium-sized minimum resource requirements
+
+### Querying and Data Visualisation
+
+The Querying and Visualization service will be supplied by Grafana, which can utilize the Storage service implementation directly as a Data Source.
+
+User interface familiarity is an important factor for developers that use the overall metrics service, so addition of other visualisation tools like Kibana that would increase the learning curve of the solution should be avoided.
+
+#### Authentication
```

Comment:
@paolobueno I think we should have something about api key auth for mobile apps calling the "thin HTTP-based service". I don't think we want those endpoints to be public.

Reply:
Sure, this one is a subheader for the "Querying and Data Visualisation" section, so I'll make it clear this relates to only acessing the grafana web ui and add another auth subheader to the "Mobile Metrics Service" section.